tools/config-generator/main_test.go

```diff
@@ -43,3 +54,849 @@ func TestOutputConfig(t *testing.T) {
+	out = newbaseProwJobTemplateData("foo/subrepo")
+	expected := "path_alias: knative.dev/subrepo"
+	if diff := cmp.Diff(out.PathAlias, expected); diff != "" {
+		t.Errorf("Unexpected path alias: (-got +want)\n%s", diff)
+	}
+
+	nonPathAliasRepos.Insert("foo/subrepo")
+	out = newbaseProwJobTemplateData("foo/subrepo")
+	if diff := cmp.Diff(out.PathAlias, ""); diff != "" {
+		t.Errorf("Unexpected path alias: (-got +want)\n%s", diff)
+	}
+
+	// don't pollute the global setup
+	pathAliasOrgs.Delete("foo")
+	nonPathAliasRepos.Delete("foo/subrepo")
+}
+
+func TestCreateCommand(t *testing.T) {
+	SetupForTesting()
+	preCommand = "" // global
+	in := baseProwJobTemplateData{Command: "foo", Args: []string{"bar", "baz"}}
+	out := createCommand(in)
+	expected := []string{"foo", "bar", "baz"}
+	if diff := cmp.Diff(out, expected); diff != "" {
+		t.Errorf("Unexpected command & args list: (-got +want)\n%s", diff)
+	}
+
+	preCommand = "expelliarmus"
+	out = createCommand(in)
+	expected = []string{"expelliarmus", "foo", "bar", "baz"}
+	if diff := cmp.Diff(out, expected); diff != "" {
+		t.Errorf("Unexpected command & args list: (-got +want)\n%s", diff)
+	}
+
+	preCommand = ""
+}
+
+func TestEnvNameToKey(t *testing.T) {
+	SetupForTesting()
+	if diff := cmp.Diff(envNameToKey("foo"), "- name: foo"); diff != "" {
+		t.Errorf("Unexpected name to key conversion: (-got +want)\n%s", diff)
+	}
+}
+
+func TestEnvValueToValue(t *testing.T) {
+	SetupForTesting()
+	if diff := cmp.Diff(envValueToValue("bar"), "  value: bar"); diff != "" {
+		t.Errorf("Unexpected env value conversion: (-got +want)\n%s", diff)
+	}
+}
+
+func TestAddEnvToJob(t *testing.T) {
+	SetupForTesting()
+	job := baseProwJobTemplateData{}
+	job.addEnvToJob("foo", "bar")
+	if diff := cmp.Diff(job.Env[0], "- name: foo"); diff != "" {
+		t.Errorf("Unexpected env name: (-got +want)\n%s", diff)
+	}
+	if diff := cmp.Diff(job.Env[1], "  value: bar"); diff != "" {
+		t.Errorf("Unexpected env value: (-got +want)\n%s", diff)
+	}
+
+	job = baseProwJobTemplateData{}
+	job.addEnvToJob("num", "42")
+	if diff := cmp.Diff(job.Env[0], "- name: num"); diff != "" {
+		t.Errorf("Unexpected env name: (-got +want)\n%s", diff)
+	}
+	if diff := cmp.Diff(job.Env[1], "  value: \"42\""); diff != "" {
+		t.Errorf("Unexpected env value: (-got +want)\n%s", diff)
+	}
+}
+
+func TestAddLabelToJob(t *testing.T) {
+	SetupForTesting()
+	job := baseProwJobTemplateData{}
+	addLabelToJob(&job, "foo", "bar")
+
+	expected := []string{"foo: bar"}
+	if diff := cmp.Diff(job.Labels, expected); diff != "" {
+		t.Errorf("Unexpected label string: (-got +want)\n%s", diff)
+	}
+}
+
+func TestAddMonitoringPubsubLabelsToJob(t *testing.T) {
+	SetupForTesting()
+	job := baseProwJobTemplateData{}
+	addMonitoringPubsubLabelsToJob(&job, "foobar")
+	expected := []string{
+		"prow.k8s.io/pubsub.project: knative-tests",
+		"prow.k8s.io/pubsub.topic: knative-monitoring",
+		"prow.k8s.io/pubsub.runID: foobar",
+	}
+	for i := range expected {
```

Comment:
Can this be replaced by a single cmp.Diff instead of for loop?

Reply:
Done